features/device_key/source/DeviceKey.cpp

```diff
@@ -270,7 +270,10 @@ int DeviceKey::generate_root_of_trust()
 
     mbedtls_entropy_free(entropy);
     delete entropy;
-    ret = device_inject_root_of_trust(key_buff, actual_size);
+
+    if (ret == DEVICEKEY_SUCCESS) {
+        ret = device_inject_root_of_trust(key_buff, actual_size);
```

Comment:
Else ? What in else case? We should probably at least to tr_error() and possibly even an assert, too?

Reply:
The "else"-Case will propagate the error code from l. 265-269.
So either ret is DEVICEKEY_SUCCESS or DEVICEKEY_GENERATE_RANDOM_ERROR at this point.